tgui/packages/tgui/interfaces/Blueprints.tsx

```diff
@@ -0,0 +1,187 @@
+import { BooleanLike } from 'common/react';
+
+import { useBackend } from '../backend';
+import { Box, Button, Divider, Section } from '../components';
+import { Window } from '../layouts';
+
+type Data = {
+  area_notice: string;
+  area_name: string;
+  wire_data: WireData[];
+  legend: string;
+  legend_viewing_list: string;
+  legend_off: string;
+  fluff_notice: string;
+  station_name: string;
+  wires_name: string;
+  wire_devices: WireDevices[];
+  viewing: BooleanLike;
+};
+
+type WireDevices = {
+  name: string;
+  ref: string;
+};
+
+type WireData = {
+  color: string;
+  message: string;
+};
+
+export const Blueprints = () => {
+  const { act, data } = useBackend<Data>();
+  const { legend, legend_viewing_list, legend_off } = data;
+
+  return (
+    <Window width={450} height={340}>
+      <Window.Content scrollable>
+        {legend === legend_viewing_list ? (
+          <WireList />
+        ) : legend === legend_off ? (
+          <MainMenu />
+        ) : (
+          <WireArea />
+        )}
+      </Window.Content>
+    </Window>
+  );
+};
+
+const WireList = () => {
+  const { act, data } = useBackend<Data>();
+  const { wire_devices = [] } = data;
+
+  return (
+    <Section>
+      <Button fluid icon="chevron-left" onClick={() => act('exit_legend')}>
+        Back
+      </Button>
+      <Box>
+        {wire_devices.map((wire) => (
+          <Button
+            width="49.5%"
+            key={wire.ref}
+            onClick={() =>
+              act('view_wireset', {
+                view_wireset: wire.ref,
+              })
+            }
+          >
+            {wire.name}
+          </Button>
+        ))}
+      </Box>
+    </Section>
+  );
+};
+
+const WireArea = () => {
+  const { act, data } = useBackend<Data>();
+  const { wires_name, wire_data = [] } = data;
+
+  return (
+    <Section>
+      <Button fluid icon="chevron-left" onClick={() => act('view_legend')}>
+        Back
+      </Button>
+      <Box bold>{wires_name} wires:</Box>
+      {wire_data.map((wire) => (
+        <Box ml={1} m={1} key={wire.message}>
+          <span style={{ color: wire.color }}>{wire.color}</span>:{' '}
+          {wire.message}
+        </Box>
+      ))}
+    </Section>
+  );
+};
+
+const MainMenu = () => {
+  const { act, data } = useBackend<Data>();
+  const { area_notice, area_name, fluff_notice, station_name, viewing } = data;
+
+  return (
+    <Section>
+      <Box bold m={1} textAlign="center">
+        {station_name} blueprints
+      </Box>
+      <Box italic fontSize={0.9}>
```

Comment:
many boxes in this ui where i feel stack.items would be more warranted, it would also space the buttons so they don't look so conjoined

Reply:
oop i did not see the melbert review

Reply:
https://github.com/tgstation/tgstation/assets/53777086/b4e040c8-4e14-4b3b-a86a-2d41bc67e540